llvm/lib/CodeGen/VirtRegMap.cpp

```diff
@@ -598,6 +598,9 @@ void VirtRegRewriter::rewrite() {
   SmallVector<Register, 8> SuperDefs;
   SmallVector<Register, 8> SuperKills;
 
+  const bool IsValidAlloc = !MF->getProperties().hasProperty(
```

Comment:
Nit: You can avoid the double negation if the variable name is changed to `IsFailedAlloc.`

Reply:
This keeps the negation out of the loop